src/python/pants/backend/python/tasks/python_repl.py

```diff
@@ -48,6 +49,9 @@ def setup_repl_session(self, targets):
       entry_point = 'code:interact'
     pex_info = PexInfo.default()
     pex_info.entry_point = entry_point
+    python_setup = PythonSetup.global_instance()
+    setup_constraints = python_setup.interpreter_constraints
+    pex_info.add_interpreter_constraint(setup_constraints[-1])
```

Comment:
We will want to wrap lines 53-54 in an `if python_setup.get_options().is_flagged('interpreter_constraints'):` because we only want to pluck the last element off in that case. When the CLI arg is not flagged, we will be getting either compatibility or config constraints set in `python_execution_task_base`.